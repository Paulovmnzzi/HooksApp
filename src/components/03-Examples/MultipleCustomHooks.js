import React from 'react'
import useFetch from '../../hooks/useFetch'
import useCounter from '../../hooks/useCounter'

import '../02-useEffect/effects.css'

const MultipleCustomHooks = () => {

    const {counter, increment} = useCounter(1);

    const { data, loading } = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);
    // console.log(data);
    const { author, quote } = !!data && data[0]; //null -> null      !null -> true      !!null -> false 
                                                //basicamente lo que dice es que si NO tiene data, arroje false, pero si SI tiene data, muestre data[0];



    return (
        <>
            <h1>Breaking Bad Quotes!</h1>

            {
                loading ?
                    (
                        <div className='alert alert-info text-center'>
                            Loading...
                        </div>
                    ) :
                    (
                        <blockquote className='blockquote text-end'>
                            <p className='mb-0'>{quote}</p>
                            <footer className='blockquote-footer mt-2'> {author} </footer>
                        </blockquote>
                    )
            }
            <button onClick={increment}>Siguiente quote</button>

        </>
    )
}

export default MultipleCustomHooks
