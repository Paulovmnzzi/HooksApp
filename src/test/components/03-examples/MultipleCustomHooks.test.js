const { shallow } = require("enzyme")
import MultipleCustomHooks from '../../../components/03-Examples/MultipleCustomHooks'
import useCounter from '../../../hooks/useCounter';
import useFetch from '../../../hooks/useFetch'
jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Pruebas en MutipleCustomHooks', () => {
    
    beforeEach(() => {

        useCounter.mockReturnValue({
            counter: 10
        })
    })

    test('Debe mostrarse correctamente', () => {
        
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })

        
        const wrapper = shallow(<MultipleCustomHooks />)
        expect(wrapper).toMatchSnapshot();
    })
    
    test('Debe mostrar la informaciÃ³n', () => {
        
        useFetch.mockReturnValue({
            data: [{
                author: 'Paulo',
                quote: 'A la gilada ni cabida'
            }],
            loading: false,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks />)

        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.mb-0').text().trim()).toBe('A la gilada ni cabida')
        expect(wrapper.find('footer').text().trim()).toBe('Paulo')


        console.log(wrapper.html());
    })
    

})
