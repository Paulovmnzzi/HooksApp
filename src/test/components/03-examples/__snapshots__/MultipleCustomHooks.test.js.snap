// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en MutipleCustomHooks Debe mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    Breaking Bad Quotes!
  </h1>
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button>
    Siguiente quote
  </button>
</Fragment>
`;
